components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Extensions/PublishedNodesEntryModelEx.cs

```diff
@@ -35,7 +35,7 @@ public static class PublishedNodesEntryModelEx {
                 return false;
             }
 
-            if (model.OpcAuthenticationMode != that.OpcAuthenticationMode) {
+            if (model.OpcAuthenticationMode != that.OpcAuthenticationMode && that.OpcAuthenticationMode != OpcAuthenticationMode.Anonymous) {
```

Comment:
>&& that.OpcAuthenticationMode != OpcAuthenticationMode.Anonymous [](http://example.com/codeflow?start=74&length=64)

what is the usecase covered by this change?

Reply:
this cover the case where OpcAuthenticationMode is not used as filtering option.
if OpcAuthenticationMode is not explicity set, it has the default value  OpcAuthenticationMode.Anonymous

Reply:
what happens when 
model.OpcAuthenticationMode = OpcAuthenticationMode.UserName
that.OpcAuthenticationMode = OpcAuthenticationMode.Anonymous
?
they look different to me, but the code sequence will conclude they are equal.

Reply:
fixed